sources/packages/backend/apps/api/src/route-controllers/student/_tests_/e2e/student.institutions.controller.searchStudents.e2e-spec.ts

```diff
@@ -0,0 +1,470 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import { DataSource, Repository } from "typeorm";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getInstitutionToken,
+  getAuthRelatedEntities,
+  InstitutionTokenTypes,
+  authorizeUserTokenForLocation,
+} from "../../../../testHelpers";
+import {
+  createFakeInstitutionLocation,
+  saveFakeApplicationDisbursements,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import {
+  Application,
+  ApplicationStatus,
+  Institution,
+  InstitutionLocation,
+  Student,
+} from "@sims/sims-db";
+
+describe("StudentInstitutionsController(e2e)-searchStudents", () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let collegeF: Institution;
+  let collegeFLocation: InstitutionLocation;
+  let studentRepo: Repository<Student>;
+  let applicationRepo: Repository<Application>;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    appDataSource = dataSource;
+    studentRepo = dataSource.getRepository(Student);
+    applicationRepo = dataSource.getRepository(Application);
+    const { institution: collegeF } = await getAuthRelatedEntities(
+      appDataSource,
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    collegeFLocation = createFakeInstitutionLocation(collegeF);
+
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeFUser,
+      collegeFLocation,
+    );
+  });
+
+  it("Should find the student by application number when student has at least one application submitted for the institution.", async () => {
+    // Arrange
+    // Student who has application submitted to institution.
+    let student = await saveFakeStudent(appDataSource);
+    // Refresh student object to get a birthDate with date only
+    student = await studentRepo.findOne({
+      where: { id: student.id },
+      relations: { sinValidation: true },
+    });
+
+    const application = await saveFakeApplicationDisbursements(
+      appDataSource,
+      {
+        institution: collegeF,
+        institutionLocation: collegeFLocation,
+        student,
+      },
+      {
+        applicationStatus: ApplicationStatus.Submitted,
+      },
+    );
+    const endpoint = "/institutions/student/search";
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    const searchPayload = {
+      appNumber: application.applicationNumber,
```

Comment:
A duplicate application number alphanumeric with 10 digits can happen but it is such a remote possibility and there's a unique key for application number in the database that would shows us the cause in case that happened. I'm not sure it is worth to invest in that approach.

Reply:
I believe that I am missing something, what is the "unique key for application number" on DB?
Do you mean that, if we try to add two applications with the same application number, the DB will throw an error?

Besides that, about if it is worth it or not, in general, I would prefer not to build the E2E tests based on the "bet" of a remote possibility of them failing when the alternative way can be implemented with a possible quick (minutes or few hours) alternatives.

I will not block this specific PR at this moment, but I believe that the comment is worth it to think about in any upcoming one.